core/app/models/spree/stock_item.rb

```diff
@@ -8,7 +8,7 @@ class StockItem < Spree::Base
 
     validates :stock_location, :variant, presence: true
     validates :variant_id, uniqueness: { scope: [:stock_location_id, :deleted_at] }, allow_blank: true, unless: :deleted_at
-    validates :count_on_hand, numericality: { greater_than_or_equal_to: 0 }, if: :verify_count_on_hand?
+    validates :count_on_hand, numericality: { greater_than_or_equal_to: 0 }, unless: :backorderable?
```

Comment:
The negative amount usually indicates that that many items are backorderd. If we adjusted below that, that would no longer mean the same thing. It makes sense to adjust positive amounts to fill backorders but I don't think it makes sense to adjust negatively.